vignettes/example.Rmd

```diff
@@ -0,0 +1,276 @@
+
+For example, the number of individuals per $m^2$ is found by integrating the density of each characteristic at a given step over height.
+
+```{r}
+data_species_tot %>% 
+  ggplot(aes(time, individuals, colour=species)) +
+  geom_line()
+```
+
+We can use the function 'expand_state' to capture a large number of additional state variables (e.g.leaf area)
+
+```{r}
+results_tidy_expand <- results_tidy %>% expand_state_FF16()
+
+results_tidy_expand$species
+```
+
+In the same way as individuals above, we can have a look at the total leaf area over time by intergrating across the size (i.e. height) distribution for a given step or time. This involves finding the product of the density of each of the characteristics present multiplied by the state variable in question (i.e. leaf area) before integrating across height. 
+
+```{r}
+data_species_tot <- 
+  results_tidy_expand$species %>% 
+  integrate_over_size_distribution()
+
+data_species_tot %>% 
+  ggplot(aes(time, area_leaf, colour = species)) +
+  geom_line()
+```
+
+
+In this case, species 1 with cheap, thin leaves briefly has a much higher total leaf area deployed comapred across the height distribution before eventually being overtaken by species with thicker, more expensive leaves. 
+
+
+```{r}
+data_species_tot %>% 
+  ggplot(aes(time, area_leaf, colour = species)) +
+  geom_line()
+```
+
+We can convert these integrated total state variables into their respective biomass components to get an idea of how total biomass and relative biomass allocation varies through time for each species. 
+
+```{r}
+v <- c("mass_heartwood", "mass_sapwood", "mass_bark", "mass_leaf")
+
+data_long <- 
+  data_species_tot %>% 
+  select(time,species, one_of(v)) %>%
+  pivot_longer(cols=starts_with("mass"), names_to = "tissue")
+
+data_long$tissue <- factor(data_long$tissue, levels = v)
+
+species_names <- tibble(species = unique(data_long$species),
+                      species_name = c("Species_1","Species_2"))
+
+data_long %>%
+  left_join(species_names) %>%
+  ggplot(aes(time, value, fill=tissue)) +
+  geom_area() +
+  labs(x = "Patch age (yr)", y = "Above ground mass (kg/m2)") +
+  theme_classic() + 
+  xlim(c(0,100)) +
+  facet_wrap(~species_name)
+```
+
+
+
+## Interpolate to specific times
+
+The cohort spacing algorithim determines the optimum introduction times for characterstics, which means that the time values associated with steps may not correspond to known times that we might like to extract information about. For example, what the total amount of biomass per $m^2$ be at 50 years. By doing this, it also means that we can integrate between defined times.
+
+
+
+```{r}
+results <- run_mypatch(traits =  trait_matrix(c(0.07, 0.24), c("lma")))
+
+results_tidy <- results %>%
+  tidy_patch()
+
+```
+
+
+Lets interpolate our results from the `run_mypatch` above for timepoints 1,5,10,1000 using the funcion `interpolate_to_times`. Note that time in units of years. 
+
+```{r}
+times <- c(1, 5, 10, 1000)
```

Comment:
I get `NA`'s for times longer than the simulation, maybe change `1000` -> `100`

Reply:
Thanks. That was intentional, 1000 being outside the range.